tendrl/commons/jobs/__init__.py

```diff
@@ -56,24 +59,37 @@ def _process_job(self, raw_job, job_key):
     def _acceptor(self):
         while not self.job_consumer_thread._complete.is_set():
             gevent.sleep(2)
-
             # TODO(team) replace below raw write with a "EtcdJobQueue" class
             try:
                 jobs = tendrl_ns.etcd_orm.client.read("/queue")
             except etcd.EtcdKeyNotFound:
                 continue
-                
             for job in jobs.children:
-                executed = False
-                if job.value is None:
+                try:
+                    raw_job = {"job_id": None,
+                               "integration_id": None,
+                               "run": None,
+                               "status": None,
+                               "parameters": None,
+                               "type": None,
+                               "node_ids": None,
+                               "request_id": None,
+                               "errors": None,
+                               "parent": None}
+                    job_id = job.key.split("/")[-1]
+                    result = self.read_job(job.key)
+                    result['job_id'] = job_id
+                    for item in result:
+                        raw_job[item] = result[item]
+                except etcd.EtcdKeyNotFound:
                     continue
-                raw_job = json.loads(job.value.decode('utf-8'))
+                executed = False
                 try:
                     # If current node's id does not fall under job's node_ids,
                     # ignore the job and dont process
                     if "node_ids" in raw_job:
                         if tendrl_ns.node_context.node_id \
-                                not in raw_job['node_ids']:
+                                not in raw_job['node_ids'].split(", "):
```

Comment:
@shtripat yes space is required for example:
    "1234-133-13-131, 1213-343-343-34"   if i give split(",") then result is ["1234-133-13-131", 
" 1213-343-343-34"] second one have space at front. to avoid this i gave like this.